scribus/plugins/import/pdf/slaoutput.cpp

```diff
@@ -3896,3 +3926,680 @@ bool SlaOutputDev::checkClip()
+	text_node->setAttribute("transform", transform);
+	g_free(transform);
+	*/
+	/*set the default charstyle to the style of the glyph, this needss fleshing out a little */
+
+	int shade = 100;
+	//TODO: This needs to come from the framework
+	//QString CurrColorText = getColor(state->getFillColorSpace(), state->getFillColor(), &shade);
+	//TODO: replace this with the framework
+	//applyTextStyleToCharStyle(pStyle.charStyle(), _glyphs[0].style->getFont().family(), CurrColorText, _glyphs[0].style->getFont().pointSizeF());// *_font_scaling);	
+	CharStyle& cStyle = static_cast<CharStyle&>(pStyle.charStyle());
+	cStyle.setScaleH(1000.0);
+	cStyle.setScaleV(1000.0);
+	cStyle.setHyphenChar(SpecialChars::BLANK.unicode());
+
+	textNode->itemText.setDefaultStyle(pStyle);
+	textNode->invalid = true;
+	activeTextRegion->renderToTextFrame(textNode, pStyle);
+	//FIXME: Paragraphs need to be implemented properly  this needs to be applied to the charstyle of the default pstyle
+	textNode->itemText.insertChars(SpecialChars::PARSEP, true);
+
+	//Set the shape so we don't clip all the text away.
+	FPointArray boundingBoxShape;
+	boundingBoxShape.resize(0);
+	boundingBoxShape.svgInit();
+
+	double bbosdoubles[32] = { 0,0
+							,0,0
+							,100,0
+							,100,0
+							,100,0
+							,100,0
+							,100,100
+							,100,100
+							,100,100
+							,100,100
+							,0,100
+							,0,100
+							,0,100
+							,0,100
+							,0,0
+							,0,0
+	};
+	boundingBoxShape.svgMoveTo(bbosdoubles[0], bbosdoubles[1]);
+	for (int a = 0; a < 16; a += 2)
+	{
+		boundingBoxShape.append(FPoint(bbosdoubles[a * 2], bbosdoubles[a * 2 + 1]));
+	}
+	boundingBoxShape.scale(textNode->width() / 100.0, textNode->height() / 100.0);
+
+	textNode->SetFrameShape(32, bbosdoubles);
+	textNode->ContourLine = textNode->PoLine.copy();
+
+	m_doc->Items->removeLast();
+	m_Elements->append(textNode);
+	if (m_groupStack.count() != 0)
+	{
+		m_groupStack.top().Items.append(textNode);
+		applyMask(textNode);
+	}
+}
+
+/*code mostly taken from importodg.cpp which also supports some line styles and more fill options etc...*/
+//FIXME: This needs to be implemented based on the framework
+void SlaOutputDev::finishItem(PageItem* item)
+{
+	item->ClipEdited = true;
+	item->FrameType = 3;
+	//this requires that PoLine is set
+	//FPoint wh = getMaxClipF(&item->PoLine);
+	//item->setWidthHeight(wh.x(), wh.y());
+	//item->Clip = flattenPath(item->PoLine, item->Segments);
+	item->OldB2 = item->width();
+	item->OldH2 = item->height();
+	item->updateClip();
+	item->OwnPage = m_doc->OnPage(item);
+	//item->setFillTransparency(1.0 - state->getFillOpacity() > state->getStrokeOpacity() ? state->getFillOpacity() : state->getStrokeOpacity());
+	//item->setLineTransparency(1.0);
+}
+
+TextFramework::TextFramework()
+{
+	m_textRegions.push_back(activeTextRegion);
+	setCharMode(AddCharMode::ADDFIRSTCHAR);
+}
+
+TextFramework::~TextFramework()
+{
+}
+
+void TextFramework::addNewTextRegion()

```

Comment:
... `addTextRegion`... if you add it it's new : - )

Reply:
lol